tests/rptest/remote_scripts/stream_verifier_txn.py

```diff
@@ -0,0 +1,1525 @@
+        "msg_total": 256
+    }
+
+    If Consume is active, returns error JSON
+    Validation errors are returned in the same way
+    {
+        "errors": [
+            "Active consume job not finished"
+        ]
+    }
+
+    """
+    def on_post(self, req: falcon.Request, resp: falcon.Response):
+        self.wlogger.debug("Processing atomic consume/produce post request")
+        # Validate request
+        if not self._validate_media_type(req, resp):
+            self.wlogger.warning("Media type invalid in atomic request")
+            return
+        elif self._check_active_thread(self.consume_thread, req, resp):
+            self.wlogger.warning("Atomic action is active")
+            return
+
+        # update topic config
+        topics_cfg_keys = list(vars(self.live_topic).keys())
+        topics_cfg_keys += ["msg_rate_limit", "total_messages"]
+        if not self._validate_request(topics_cfg_keys, [], req, resp):
+            # response is already populated inside validate_request
+            return
+        elif req.media['source_topic_name'] == req.media['target_topic_name']:
+            self._http_400(["Source and taget topics can't be equal"], resp)
+            return
+        else:
+            # Update topic configs
+            for k, v in req.media.items():
+                self.live_topic.__setattr__(k, v)
+            # start consumers
+            self.init_atomic_produce()
+            self.atomic_produce(wait=False)
+
+            resp.status = falcon.HTTP_200
+            resp.content_type = falcon.MEDIA_JSON
+            resp.media = {'result': "OK"}
+
+
+def start_webserver():
+    def terminate_handler(signum, frame):
+        """SIGTERM handler
+        """
+        signame = signal.Signals(signum).name
+        logger.info(f"{signame} ({signum}) received, terminating threads")
+        producer.terminate()
+        consumer.terminate()
+
+    def add_route(route: str, handler: AppCfg | StreamVerifier):
+        logger.debug(f"Registering handler for '/' as {type(handler)}")
+        app.add_route(route, handler)
+
+    global app_config
+    logger = setup_logger(LOGGER_STARTUP)
+
+    app = falcon.App()
+    json_handler = falcon.media.JSONHandler(dumps=partial(
+        json.dumps,
+        cls=NpEncoder,
+        sort_keys=True,
+    ), )
+    extra_handlers = {
+        'application/json': json_handler,
+    }
+
+    app = falcon.App()
+    app.req_options.media_handlers.update(extra_handlers)
+    app.resp_options.media_handlers.update(extra_handlers)
+
+    # Add subpages
+    logger.debug("Initializing producer class")
+    producer = StreamVerifierProduce(app_config)
+    logger.debug("Initializing consumer class")
+    consumer = StreamVerifierConsume(app_config)
+    logger.debug("Initializing atomic class")
+    atomic = StreamVerifierAtomic(app_config)
+    add_route("/", app_config)
+    add_route("/produce", producer)
+    add_route("/consume", consumer)
+    add_route("/atomic", atomic)
```

Comment:
suggestion on naming /consume -> /verify

I think /produce and /atomic can be merged into one /start or something (we want to start produce and consume at the same time, the split is not needed from a user perspective (feel free to skip if too much work)

Reply:
I still want to do produce as a separate step since combining them will introduce complications into future debugging.

Agree on consume -> verify renaming

Reply:
Done.

Reply:
> Done.

I don't see the change?

Reply:
Yes, I was updating the test and debugging collateral PR. It is updated now.